uktrade/data-hub-frontend
# Implement per-user feature flags and refactor layout-testing

## Description of change

Consumes the new per-user feature flags from the api to toggle the new dashboard layout. The existing per-team layout feature testing has been retained, but refactored to make it a bit more robust. Layout testing was previously unable to cope with multiple teams using the layout testing feature flag - this now checks each of them.

## Test instructions

- Assign the "personalised-dashboard" feature flag to your user through Django Admin (under company/adviser)
- You should now be able to see the personalised dashboard
- Un-setting the flag for your user should revert to the old dashboard (assuming you are not in a team with the feature enabled)

- Set multiple `layoutTesting` feature flags in django admin (using the format `layoutTesting:<team_id>`) - check that the personalised dashboard is visible when you are a member of one of the testing teams and the old dashboard when you are not (assuming your user does not have the flag set).

The acceptance criteria give each of the different testing scenarios here: https://trello.com/c/3SblPrH8/551-enable-per-user-dashboard-feature-flag

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
